builtin/providers/rabbitmq/resource_binding.go

```diff
@@ -0,0 +1,160 @@
+package rabbitmq
+
+import (
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/michaelklishin/rabbit-hole"
+
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceBinding() *schema.Resource {
+	return &schema.Resource{
+		Create: CreateBinding,
+		Update: nil,
+		Read:   ReadBinding,
+		Delete: DeleteBinding,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"source": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"vhost": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"destination": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"destination_type": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"properties_key": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"routing_key": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+
+			"arguments": &schema.Schema{
+				Type:     schema.TypeMap,
+				Optional: true,
+				ForceNew: true,
+			},
+		},
+	}
+}
+
+func CreateBinding(d *schema.ResourceData, meta interface{}) error {
+	rmqc := meta.(*rabbithole.Client)
+	vhost := d.Get("vhost").(string)
+
+	bindingInfo := rabbithole.BindingInfo{
+		Source:          d.Get("source").(string),
+		Destination:     d.Get("destination").(string),
+		DestinationType: d.Get("destination_type").(string),
+		RoutingKey:      d.Get("routing_key").(string),
+		PropertiesKey:   d.Get("properties_key").(string),
+		Arguments:       d.Get("arguments").(map[string]interface{}),
+	}
+
+	if err := declareBinding(rmqc, vhost, bindingInfo); err != nil {
+		return err
+	}
+
+	name := fmt.Sprintf("%s/%s/%s/%s/%s", vhost, bindingInfo.Source, bindingInfo.Destination, bindingInfo.DestinationType, bindingInfo.PropertiesKey)
+	d.SetId(name)
+
+	return nil
+}
+
+func ReadBinding(d *schema.ResourceData, meta interface{}) error {
```

Comment:
I think the Read should check for 404 and remove from state if not found - this will allow us to avoid errors when someone deletes a binding manually